### DISCUSSION TITLE
zwave-js-ui for ARM64 pimox for Z-Pi 7 GPIO board

### DISCUSSION THREAD

**Discussion by Wri84** [tteck/Proxmox#637] [Help Needed]
Hi,

I'm trying to run Homeassistant supervised in VM on pimox.

I've currently installed a Z-PI 7 GPIO board, which  when running HA baremetal, works just fine. However, I'm unable to passthrough the dev/ttyAMA0 device to the VM.

So as a workaround, I wanted to run zwave JS in LXC on the RPI and detach homeassistant from Zwave. However, the script provided seems to be x86 only and also doesn't start, since ttyAMA0 is not part of the mountpoints

```
lxc.mount.entry: /dev/serial/by-id  dev/serial/by-id  none bind,optional,create=dir
lxc.mount.entry: /dev/ttyUSB0       dev/ttyUSB0       none bind,optional,create=file
lxc.mount.entry: /dev/ttyUSB1       dev/ttyUSB1       none bind,optional,create=file
lxc.mount.entry: /dev/ttyACM0       dev/ttyACM0       none bind,optional,create=file
lxc.mount.entry: /dev/ttyACM1       dev/ttyACM1       none bind,optional,create=file
```

So, could someone either guide me how to passthrough the ttyAMA0 correctly to the HAOS VM, or otherwise tell me how I can run Zwave-JS-UI in LXC on Pimox?

Help would be greatly appreciated!

**Comment by tteck** [comment]
This was forwarded to me from @davidmourenza over at https://community.home-assistant.io/
I have no Pi, so he did a lot of testing for me.

> Turn off the Home Assistant VM machine.
Go to the pimox host shell.
`lsusb`
The usbs attached will appear with numbers like this:
```
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 005: ID 10c4:ea60 Silicon Labs CP210x UART Bridge
Bus 001 Device 004: ID 1915:cafe Nordic Semiconductor ASA nRF528xx OpenThread Device
Bus 001 Device 003: ID 10c4:ea60 Silicon Labs CP210x UART Bridge
Bus 001 Device 002: ID 2109:3431 VIA Labs, Inc. Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```
> We only need the hexadecimal codes, before and after the :
For example for Nordic Semiconductor we need 1915:cafe
And we need to add this code line:
`args: -device qemu-xhci -device usb-host,vendorid=0xVendorid,productid=0xProductid`
changing the vendor id and product id with the hexadecimal digits.
So the correct is:
`args: -device usb-host,vendorid=0x1915,productid=0xcafe`
If you have many usb devices you only need to add ones args and the write the rest, so will be:
`args: -device qemu-xhci -device usb-host,vendorid=0x10c4,productid=0xea60 -device qemu-xhci -device usb-host,vendorid=0x10c4,productid=0xea60 -device qemu-xhci -device usb-host,vendorid=0x1915,productid=0xcafe`
This line need to be added here

> `sudo nano /etc/pve/qemu-server/101.conf `(101 or the number of the home assistant machine).
Add the line after all the other lines.
Don’t use the proxmox ui to use usb or the VM never will come up.

**Comment by tteck** [comment]
If you need his help... https://community.home-assistant.io/t/ha-os-on-pimox-proxmox-v7-for-rpi/406149/17?u=tteck

**Reply by Wri84** [reply]
Thanks for the fast response as always :) I've used your script for HAOS on Pimox.

The issue with my z-wave device is, that it is not USB but serial connected to the GPIO pins of the pi. Therefore it's not listed in LSUSB, but mounted under dev/ttyAMA0. 

This is what is causing the problem, because I don't know how to pass it through to the VM. I believe an LXC container would have direct access, which is why I wanted to go down the decoupled zwave-js-ui route
